flytekit/core/type_engine.py

```diff
@@ -407,17 +407,26 @@ def literal_map_to_kwargs(
         return {k: TypeEngine.to_python_value(ctx, lm.literals[k], v) for k, v in python_types.items()}
 
     @classmethod
-    def dict_to_literal_map(cls, ctx: FlyteContext, d: typing.Dict[str, typing.Any]) -> LiteralMap:
+    def dict_to_literal_map(
+        cls, ctx: FlyteContext, d: typing.Dict[str, typing.Any], guessed_python_types: typing.Dict[str, type] = None
```

Comment:
```suggestion
        cls, ctx: FlyteContext, d: typing.Dict[str, typing.Any], guessed_python_types: typing.Optional[typing.Dict[str, type]] = None
```
but don't use github to take the suggestion - it doesn't do dco.